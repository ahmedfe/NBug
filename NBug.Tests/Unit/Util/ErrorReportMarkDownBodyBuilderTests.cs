// --------------------------------------------------------------------------------------------------------------------
// <copyright file="GitHubUrlBuilderTests.cs" company="Git Extensions">
//   Copyright (c) 2019 Igor Velikorossov. Licensed under MIT license.
// </copyright>
// --------------------------------------------------------------------------------------------------------------------

using System;
using ApprovalTests;
using FluentAssertions;
using NBug.Core.Util;
using Xunit;

namespace NBug.Tests.Unit.Util
{
    public class ErrorReportMarkDownBodyBuilderTests
    {
        private readonly ErrorReportMarkDownBodyBuilder _errorReportMarkDownBodyBuilder;

        public ErrorReportMarkDownBodyBuilderTests()
        {
            Settings.GetSystemInfo = null;
            _errorReportMarkDownBodyBuilder = new ErrorReportMarkDownBodyBuilder();
        }

        [Fact]
        public void Build_should_throw_ArgumentNullException_if_exception_null()
        {
            ((Action)(() => _errorReportMarkDownBodyBuilder.Build(null, null))).Should().Throw<ArgumentNullException>();
        }

        [Fact]
        public void Build_should_build_report_without_system_or_additional_info()
        {
            try
            {
                Mock.DoWork();

                Assert.True(false, "Expected an exception to be thrown");
            }
            catch (Exception ex)
            {
                var report = _errorReportMarkDownBodyBuilder.Build(ex, null);
                Approvals.Verify(report);
            }
        }

        [Fact]
        public void Build_should_build_report_with_system_without_additional_info()
        {
            Settings.GetSystemInfo = () => @"
- Git Extensions 0.0.2.5232
- Build 23b6f51905006ccdde8701591df706284b4155dc
- Git 2.19.0.windows.1
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.7.3324.0
- DPI 144dpi (150% scaling)
";

            try
            {
                Mock.DoWork();

                Assert.True(false, "Expected an exception to be thrown");
            }
            catch (Exception ex)
            {
                var report = _errorReportMarkDownBodyBuilder.Build(ex, null);
                Approvals.Verify(report);
            }
        }

        [Fact]
        public void Build_should_build_report_with_system_and_additional_info()
        {
            try
            {
                Mock.DoWork();

                Assert.True(false, "Expected an exception to be thrown");
            }
            catch (Exception ex)
            {
                var report = _errorReportMarkDownBodyBuilder.Build(ex, "It happened when I was arguing with someone on the Internet.\r\nI was right and they were absolutely wrong!\r\nI win! But the app crashed!\r\nPLEASE HELP!!!");
                Approvals.Verify(report);
            }
        }
    }
}