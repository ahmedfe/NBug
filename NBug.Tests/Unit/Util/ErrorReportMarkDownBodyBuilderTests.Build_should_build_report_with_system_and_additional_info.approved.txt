<!--
    :warning: Review existing issues to see whether someone else has already reported your issue.
-->

## Current behaviour

<!-- Be as specific and detailed as possible to help us identify your issue. -->


## Expected behaviour


## Steps to reproduce

<!-- Take some time to try and reproduce the issue, then explain how to do so here. -->


## Error Details
```
System.ApplicationException: Failed ---> System.ArgumentOutOfRangeException: Opps ---> System.DivideByZeroException: Boom!
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at NBug.Tests.Unit.Util.Mock.Method1() in C:\Development\gitextensions\Externals\NBug\NBug.Tests\Unit\Util\Mock.cs:line 13
   at NBug.Tests.Unit.Util.Mock.Method2() in C:\Development\gitextensions\Externals\NBug\NBug.Tests\Unit\Util\Mock.cs:line 16
   at NBug.Tests.Unit.Util.Mock.Method3() in C:\Development\gitextensions\Externals\NBug\NBug.Tests\Unit\Util\Mock.cs:line 17
   at NBug.Tests.Unit.Util.Mock.Method4() in C:\Development\gitextensions\Externals\NBug\NBug.Tests\Unit\Util\Mock.cs:line 18
   at NBug.Tests.Unit.Util.Mock.DoWork() in C:\Development\gitextensions\Externals\NBug\NBug.Tests\Unit\Util\Mock.cs:line 7
   at NBug.Tests.Unit.Util.ErrorReportMarkDownBodyBuilderTests.Build_should_build_report_with_system_and_additional_info() in C:\Development\gitextensions\Externals\NBug\NBug.Tests\Unit\Util\ErrorReportMarkDownBodyBuilderTests.cs:line 77
```


## Additional information
It happened when I was arguing with someone on the Internet.
I was right and they were absolutely wrong!
I win! But the app crashed!
PLEASE HELP!!!


## Environment
System information is not supplied
